plugins/qcheck-stm/src/stm_of_ir.ml

```diff
@@ -335,6 +335,42 @@ let may_raise_exception v =
   | [], [] -> false
   | _, _ -> true
 
+let big_product xs =
```

Comment:
Two suggestions there:
- this could be renamed `list_and`,
- it would be nicer to perform the fold (fold right, rather) in our code rather than in the generated one, so that the first false condition terminates the evaluation (I think it would be also more legible in the end)